### PR TITLE
added visual basic cheat sheet

### PR DESCRIPTION
**Pull request by shymary**
added visual basic cheat sheet
#2127 

-------
Ia page:  https://duck.co/ia/view/visual_basic_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@shymary Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
@shymary Thanks for the PR, however I think this actually more like a Visual Studio cheatsheet? Can we add some information about the actual programming language?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by gautamkrishnar** [comment]
@shymary mam, Welcome to the DuckDuckGo community. We are happy to have you in our team. 

I think this is the first ever pull request for DuckDuckGO from a university professor :smile: 

cc: @tagawa @moollaza 
@mintsoft this is actually a visual studio IDE cheatsheet. Not for VS language

**Comment by mintsoft** [comment]
@shymary @gautamkrishnar In which case let's rename it to `Visual Studio Cheatsheet`! :+1:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/visual_basic.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    "section_order":[
+        "GENERAL",
+        "CODE MODULE",
+        "PROJECT EXPLORER"
+    ],
+    "sections":{
+        "GENERAL":[
+            {
+                "val":"This toggles switch displays the VBE from the application window and vice versa",
+                "key":"Alt + F11"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/visual_basic.json]: Please remove the `+` for all your `key`s.

Each keyboard key needs to be wrapped in `[ ]`. E.g. `[Alt] [F11]`

Once that's updated the look of the cheat sheet will be much better. It should look like this: https://duckduckgo.com/?q=vim+cheat+sheet&t=ffab&ia=cheatsheet

File: share/goodie/cheat_sheets/json/visual_basic.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            },
+            {
+                "val":"GoES TO END MODULE",
+                "key":"Ctrl + End"
+            }
+        ],
+        "PROJECT EXPLORER":[
+            {
+                "val":"Selects the first file in the list",
+                "key":"HOME"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/visual_basic.json]: These ones are correct, single keys do not need `[ ]` but the key names should be consistently styled. Let's use Title Case: e.g. `Home`, `End`

For arrow keys, please use the ASCII arrow icons in the documentation.

docs.duckduckhack.com

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@shymary Please rename the file to `visual-basic.json`.

**Comment by moollaza** [comment]
@shymary Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by gautamkrishnar** [comment]
@moollaza sorry for the delay. I think she's busy with her works. I will continue this PR.

**Comment by alohaas** [comment]
@shymary thanks for trying to create a new Visual Basic Cheat Sheet, sorry it wasn't merged, but you did a great job! We'd like you to help our community tackle the next big challenge – our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements 

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

**Pull request closed without merging**